soccer/gameplay/tests/test_shooting.py

```diff
@@ -0,0 +1,99 @@
+import unittest 
+import robocup
+import main
+import evaluation.shooting
+import constants
+
+class TestShooting(unittest.TestCase):
+	def __init__(self, *args, **kwargs):
+		super(TestShooting, self).__init__(*args, **kwargs)
+		self.system_state = robocup.SystemState()	
+
+	def setUp(self):
+		main.set_system_state(self.system_state)
+
+		for robot in main.system_state().their_robots:
+			robot.set_vis_for_testing(True)
+
+		self.length = constants.Field.Length
+		self.width = constants.Field.Width
+		self.botRadius = constants.Robot.Radius
+
+		self.their_robots = main.system_state().their_robots[0:6]
+		self.our_robots = main.system_state().our_robots[0:6]
+
+		self.success = 1
+		self.failure = 0
+
+
+
+	def eval_shot(self, x, y, excluded_robots=[]):
+		return evaluation.shooting.eval_shot(robocup.Point(x, y), excluded_robots)
+
+
+	# Set the location of a robot 
+	# We must use this function so that the C++ can act on the robot location
+	# 
+	# @param a_bot: bot to change position
+	# @param x: new x position of bot
+	# @param y: new y position of bot
+	#
+	def set_bot_pos(self, a_bot, x, y):
+		a_bot.set_pos_for_testing(robocup.Point(x, y))
+
+	def test_eval_shot_clear_field(self):
+		# Test shot from their goal (to their goal)
+		self.assertEqual(self.eval_shot(0, self.length), self.success)
+		
+		# Test shot from 3/4 of the field 
+		self.assertEqual(self.eval_shot(0, 3 * self.length / 4), self.success)
+
+		# Test shot from just past the center of the field
+		self.assertEqual(self.eval_shot(0, self.length / 2 + self.botRadius), self.success)
+
+		# Test shot from some angles
+		self.assertGreater(self.eval_shot(self.width / 4, 3 * self.length / 4), 0.99)
+		self.assertGreater(self.eval_shot(-self.width / 4, 3 * self.length / 4), 0.99)
+
+		self.assertGreater(self.eval_shot(self.width / 4, self.length / 2 + self.botRadius), 0.95)
+		self.assertGreater(self.eval_shot(-self.width / 4, self.length / 2 + self.botRadius), 0.95)
+
+	@unittest.skip("Skip Problematic Cases")
+	def test_eval_shot_problem_cases(self):
+		# This location is very outside the field but we consider it successful
+		# This case return almost 1 when it should fail
+		self.assertEqual(self.eval_shot(0, 2 * self.length), self.failure)
+
+		# Does not allow shot from midfield
+		# This case returns 0 
+		self.assertEqual(self.eval_shot(0, self.length / 2), self.success)
```

Comment:
I think this is ok. We cannot score from our side of the field, so we just return a 0% shot chance when we try and evaluate it.